mail/backup_mailer/src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -156,14 +162,20 @@ public function sendEmail($config, $confdata)
         $mail->Subject = $hostname . ' OPNsense config backup ' . $date;
         $mail->Body    = $hostname . ' config backup file';
 
+        if ((string)$config->SelfSigned === "1") {
+            $selfSigned = true;
+        } else {
+            $selfSigned = false;
+        }
+
         if ((string)$config->SmtpTLS === "1") {
             $mail->SMTPSecure = 'tls';
         } else {
             $mail->SMTPOptions = array(
                 'ssl' => array(
-                    'verify_peer' => false,
+                    'verify_peer' => true,
                     'verify_peer_name' => false,
```

Comment:
with this setting, it accepts any valid certificate, without checking if the hostname matches. It does still allow MITM.
You may use the inverted value if $selfSigned here as this settings will probably never match.

@fichtner what do you think?

https://secure.php.net/manual/de/context.ssl.php

Reply:
yes but it means that in that case TLS is disabled anyway. In this case we explicitly enable TLS for a connection so a user expects it to be
1) confidential (securely encrypted to the other end)
2) authenticated (talking to the right server and not NATed by an attacker

If somebody messes with the connection, it should be dropped in my opinion.

Reply:
What's the difference between SMTPOptions and SMTPSecure ?

Reply:
FWIW, the problem will be home users who don't get that stuff right and ask questions because their mail server is not correctly set up.

Reply:
@fichtner smtpoptions seem to go directly into the socket context which is used by PHP to connect and the other is a library internal if it should use TLS/SSL.

Reply:
so we are talking about the case where it *may* use TLS, not enforce it?